root_orchestrator/resource-abstractor/db/apps_db.py

```diff
@@ -0,0 +1,38 @@
+import db.mongodb_client as db
+from bson.objectid import ObjectId
+
+
+def find_apps(filter={}):
+    return db.mongo_apps.find(filter)
+
+
+def find_app_by_id(app_id, extra_filter={}):
+    filter = {**extra_filter, "_id": ObjectId(app_id)}
+    app = list(find_apps(filter=filter))
+
+    return app[0] if app else None
+
+
+def delete_app(app_id):
+    filter = {"_id": ObjectId(app_id)}
+    return db.mongo_apps.find_one_and_delete(filter, return_document=True)
+
+
+def update_app(app_id, data):
+    return db.mongo_apps.find_one_and_update(
+        {"_id": ObjectId(app_id)},
+        {"$set": data},
+        return_document=True,
+    )
+
+
+def _create_app(app_data):
```

Comment:
Why is this separate helper function needed? Correct me if I am wrong, but isn't it only used in the method below it so couldn't we simply fuse them?

Reply:
You'll totally right! There was a purpose where `create` actually would have a flag called `upsert` & `_create` is only strict to only creating, but I didn't proceed with that and I forgot to clean it up.